repondeur/zam_repondeur/views/lectures.py

```diff
@@ -0,0 +1,185 @@
+import os
+import re
+from dataclasses import dataclass
+from tempfile import NamedTemporaryFile
+from typing import Any, Generator, List
+
+from pyramid.httpexceptions import HTTPBadRequest, HTTPFound
+from pyramid.request import Request
+from pyramid.response import FileResponse, Response
+from pyramid.view import view_config, view_defaults
+
+from zam_aspirateur.amendements.models import Amendement
+from zam_aspirateur.amendements.fetch import fetch_and_parse_all, NotFound
+from zam_aspirateur.amendements.writer import write_csv, write_xlsx
+from zam_aspirateur.__main__ import process_amendements
+
+
+CHAMBRES = {"assemblee": "Assemblée nationale", "senat": "Sénat"}
+
+SESSIONS = ["2017-2018"]
+
+
+RE_TEXTE = re.compile(r"^(?P<chambre>\w+)-(?P<session>\d{4}-\d{4})-(?P<num_texte>\d+)$")
+
+
+@dataclass
+class Lecture:
+    chambre: str
+    session: str
+    num_texte: str
+
+    @property
+    def chambre_disp(self) -> str:
+        return CHAMBRES[self.chambre]
+
+    def __str__(self) -> str:
+        return f"{self.chambre_disp}, session {self.session}, texte nº {self.num_texte}"
+
+    def __lt__(self, other: Any) -> bool:
+        if type(self) != type(other):
+            return NotImplemented
+        return (self.chambre, self.session, int(self.num_texte)) < (
+            other.chambre,
+            other.session,
+            int(other.num_texte),
+        )
+
+
+@view_config(route_name="lectures_list", renderer="lectures_list.html")
+def lectures_list(request: Request) -> dict:
+    return {
+        "lectures": list(
+            sorted(
```

Comment:
I wonder which one is the more pertinent here between sorted by num text or by date of insertion.

Reply:
My guess is that people would typically be working on the most recent law, but we can easily change that based on user feedback.